torchtitan/metrics.py

```diff
@@ -93,26 +100,55 @@ def build_device_memory_monitor():
         f"{device_type.upper} capacity: {device_memory_monitor.device_name} ({device_memory_monitor.device_index}) "
         f"with {device_memory_monitor.device_capacity_gib:.2f}GiB memory"
     )
-
     return device_memory_monitor
 
 
 class MetricLogger:
-    def __init__(self, log_dir, tag, enable_tb):
+    def __init__(self, log_dir, tag, enable_tb, enable_wandb=False, wandb_config=None):
         self.tag = tag
         self.writer: Optional[SummaryWriter] = None
-        if enable_tb:
+        self.use_wandb = False
+
+        if enable_wandb and WANDB_AVAILABLE:
+            self.use_wandb = True
+            if wandb.run is None:
+                project_name = (
+                    wandb_config.get("project", "torchtitan")
+                    if wandb_config
+                    else "torchtitan"
+                )
```

Comment:
Since you've specify `torchtitan` as default project in `config_manager.py`, no need to test if it's empty?